packages/entity-server/src/api/EntityApi.ts

```diff
@@ -0,0 +1,428 @@
+    hierarchyName: string,
+    entityCode: string,
+    queryOptions?: {
+      field: FlattableEntityFieldName;
+      filter?: EntityFilterObject;
+    },
+  ): Promise<FlattenedEntity[]>;
+  public async getRelativesOfEntity<T extends ExtendedEntityFieldName[]>(
+    hierarchyName: string,
+    entityCode: string,
+    queryOptions?: {
+      fields: T;
+      filter?: EntityFilterObject;
+    },
+  ): Promise<EntityApiResponse<T>[]>;
+  public async getRelativesOfEntity<T extends ExtendedEntityFieldName[]>(
+    hierarchyName: string,
+    entityCode: string,
+    queryOptions?: {
+      field?: FlattableEntityFieldName;
+      fields?: T;
+      filter?: EntityFilterObject;
+    },
+  ): Promise<FlattenedEntity[] | EntityApiResponse<T>[]> {
+    return this.outboundConnection.get(
+      await this.authHander.getAuthHeader(),
+      this.baseUrl,
+      `hierarchy/${hierarchyName}/${entityCode}/relatives`,
+      {
+        ...this.stringifyQueryParameters(queryOptions),
+      },
+    );
+  }
+
+  public async getRelativesOfEntities(
+    hierarchyName: string,
+    entityCodes: string[],
+    queryOptions?: {
+      field: FlattableEntityFieldName;
+      filter?: EntityFilterObject;
+    },
+  ): Promise<FlattenedEntity[]>;
+  public async getRelativesOfEntities<T extends ExtendedEntityFieldName[]>(
+    hierarchyName: string,
+    entityCodes: string[],
+    queryOptions?: {
+      fields: T;
+      filter?: EntityFilterObject;
+    },
+  ): Promise<EntityApiResponse<T>[]>;
+  public async getRelativesOfEntities<T extends ExtendedEntityFieldName[]>(
+    hierarchyName: string,
+    entityCodes: string[],
+    queryOptions?: {
+      field?: FlattableEntityFieldName;
+      fields?: T;
+      filter?: EntityFilterObject;
+    },
+  ): Promise<FlattenedEntity[] | EntityApiResponse<T>[]> {
+    return this.outboundConnection.get(
+      await this.authHander.getAuthHeader(),
+      this.baseUrl,
+      `hierarchy/${hierarchyName}/relatives`,
+      {
+        ...this.stringifyQueryParameters(queryOptions),
+        entities: entityCodes.join(MULTIPLE_VALUES_DELIMITER),
+      },
+    );
+  }
+
+  public async getRelationshipsOfEntityByAncestor(
+    hierarchyName: string,
+    entityCode: string,
+    queryOptions?: EntityApiQueryOptions,
+    ancestorQueryOptions?: RelationshipsSubQueryOptions,
+    descendantQueryOptions?: RelationshipsSubQueryOptions,
+  ): Promise<GroupByAncestorRelationshipsResponseBody> {
+    return this.outboundConnection.get(
+      await this.authHander.getAuthHeader(),
+      this.baseUrl,
+      `hierarchy/${hierarchyName}/${entityCode}/relationships`,
+      {
+        ...this.stringifyQueryParameters(queryOptions),
+        ...this.stringifyRelationshipsSubQueryParameters(ancestorQueryOptions, 'ancestor'),
+        ...this.stringifyRelationshipsSubQueryParameters(descendantQueryOptions, 'descendant'),
+        groupBy: 'ancestor',
+      },
+    );
+  }
+
+  public async getRelationshipsOfEntityByDescendant(
```

Comment:
Gave different method names to each of the different `groupBy` options, but am considering that perhaps it would be best to use method overloading again...